discovery/service.go

```diff
@@ -36,12 +36,18 @@ type certHashExtractor func(ctx context.Context) []byte
 // dispatcher defines a function that dispatches a query
 type dispatcher func(q *discovery.Query) *discovery.QueryResult
 
+// Service describes service discovery methods directly accessible to other components
```

Comment:
Why do you need to define this? You can just export the method itself and define the interface where you need it.

Reply:
Noted and removed